packages/aws-cdk/lib/api/hotswap/lambda-functions.ts

```diff
@@ -15,22 +15,20 @@ export async function isHotswappableLambdaFunctionChange(
   if (typeof lambdaCodeChange === 'string') {
     return lambdaCodeChange;
   } else {
-    // verify that the Asset changed - otherwise,
-    // it's a Code property-only change,
-    // but not to an asset change
-    // (for example, going from Code.fromAsset() to Code.fromInline())
-    if (!assetMetadataChanged(change)) {
-      return ChangeHotswapImpact.REQUIRES_FULL_DEPLOYMENT;
-    }
-
     const functionName = await establishResourcePhysicalName(logicalId, change.newValue.Properties?.FunctionName, evaluateCfnTemplate);
+    const functionArn = await evaluateCfnTemplate.evaluateCfnExpression({
+      'Fn::Sub': 'arn:${AWS::Partition}:lambda:${AWS::Region}:${AWS::AccountId}:function:' + functionName,
+    });
+
```

Comment:
Let's kill this empty line (I'm actually surprised the linter didn't flag this one):

```suggestion
```